src/naive/date.rs

```diff
@@ -192,6 +192,10 @@ impl Days {
 /// [proleptic Gregorian date]: crate::NaiveDate#calendar-date
 #[derive(PartialEq, Eq, Hash, PartialOrd, Ord, Copy, Clone)]
 #[cfg_attr(feature = "rkyv", derive(Archive, Deserialize, Serialize))]
+#[cfg_attr(
+    feature = "rkyv",
+    archive_attr(derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash))
```

Comment:
Deriving those traits for the archived type pretty much makes us promise that `NaiveDate` will not be modified in a way that makes comparing the type any different from a plain integer. But I think that is a useful property anyway.

Reply:
If `DateImpl` changes you'll have to implement `Eq`/`Ord` in a way consistent with the current implementation, in which case it should be possible to implement rkyv traits and `Eq`/`Ord` on the Archived type in the same way. What I mean is, unless I'm misunderstanding, deriving traits for the Archived type doesn't make you promise anything you don't promise already.

Reply:
True. But that is not all that easy, and would amount to deserializing the `Archived` type. That seems to defeat the purpose.

Anyway I think we are good.